src/types/tuple.rs

```diff
@@ -346,6 +331,22 @@ fn type_input() -> TypeInfo {
     }
 });
 
+fn array_into_tuple<const N: usize>(py: Python<'_>, array: [PyObject; N]) -> Py<PyTuple> {
+    unsafe {
+        let ptr = ffi::PyTuple_New(N.try_into().expect("0 < N <= 12"));
+        let tup = Py::from_owned_ptr(py, ptr);
+        let mut index = 0;
```

Comment:
`for (idx, obj) in array.into_iter().enumerate()` instead of open-coding the `index` part.

Reply:
I had open coded because the index needs to be `isize`, but clippy wanted this changed too, so I've changed and just using `index as ffi::Py_ssize_t` at the callsites. (Should always be within range.)